samples/cloud-client/snippets/idtoken_from_impersonated_credentials.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [auth_cloud_idtoken_impersonated_credentials]
+
+import google
+from google.auth import impersonated_credentials
+import google.auth.transport.requests
+
+
+def idtoken_from_impersonated_credentials(
+        impersonated_service_account: str, scope: str, target_audience: str):
+    """
+      Use a service account (SA1) to impersonate as another service account (SA2) and obtain id token
+      for the impersonated account.
+      To obtain token for SA2, SA1 should have the "roles/iam.serviceAccountTokenCreator" permission
+      on SA2.
+
+    Args:
+        impersonated_service_account: The name of the privilege-bearing service account for whom the credential is created.
+            Examples: name@project.service.gserviceaccount.com
+
+        scope: Provide the scopes that you might need to request to access Google APIs,
+            depending on the level of access you need.
+            The best practice is to use the cloud-wide scope and use IAM to narrow the permissions.
```

Comment:
Since we talked about this with Kurtis, I became aware of some use cases for using a more narrow scope, so we've backed away from calling it a "best practice" to use the cloud-wide scope. Maybe just say "For this example, we
use the cloud-wide scope and...".

Reply:
Done, thanks.